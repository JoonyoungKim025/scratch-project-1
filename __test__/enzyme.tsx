  
import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';
import renderer from 'react-test-renderer';

// **Overview: We will be testing LOGIN, REGISTER, and chatWindow(input button) using enzyme shallow rendering
// test FUNCTIONALITY and UI rendering for each**

// Enzyme is a wrapper around React test utilities which makes it easier to
// shallow render and traverse the shallow rendered tree.
// ... import componenets
import Login from '../client/loginSignUp/login';
import RegistraionForm from '../client/loginSignUp/register';
import ChatWindow from '../client/chat/chatWindow';

// Newer Enzyme versions require an adapter to a particular version of React
configure({ adapter: new Adapter() });

describe('React unit tests', () => {

    // Test for LOGIN
    describe('Login', () => {

        it('renders correctly', () => {
            const tree = renderer.create(
              <Login />
            ).toJSON();
            expect(tree).toMatchSnapshot();
          });
    })

});