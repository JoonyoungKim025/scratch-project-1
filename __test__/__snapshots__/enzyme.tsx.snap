// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React unit tests Login renders correctly 1`] = `
<div
  className="login"
>
  <div>
    <form
      className="loginSignUpFields"
    >
      <div>
        <input
          id="email"
          onChange={[Function]}
          placeholder="Enter email"
          type="email"
          value=""
        />
      </div>
      <div>
        <input
          id="password"
          onChange={[Function]}
          placeholder="Enter password"
          type="password"
          value=""
        />
      </div>
      <button
        className="loginSignUpButtons"
        onClick={[Function]}
        type="submit"
      >
        Login
      </button>
    </form>
  </div>
</div>
`;
